drivers/inc/stm32f446xx_usart_driver_course.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * stm32f446xx_usart_driver_course.h
+ *
+ *  Created on: Apr 21, 2020
+ *      Author: quentin
+ */
+#define QUENTIN
+
+#ifdef QUENTIN
+
+#ifndef INC_STM32F446XX_USART_DRIVER_COURSE_H_
+#define INC_STM32F446XX_USART_DRIVER_COURSE_H_
+
+#include "stm32f446xx_course.h"
+
+/* GPIO_PIN_CONF_TypeDef */
+typedef struct{
+	uint32_t USART_BaudRate; //size might change
+	uint8_t USART_WordLength;
+	uint8_t USART_StopBits;//0=1 bits, 1=0.5 bit, 2=2 bits, 3=1.5 bit
+	uint8_t USART_Parity;
+	uint8_t USART_Mode;
+	uint8_t USART_HardwareFlowCtrl;
+	uint8_t USART_OverSampling;
+}USART_PIN_CONF_TypeDef;
+
+typedef struct{
+	_Quentin_USART_RegDef_t* pUSART; /*holds register addresses*/
+	USART_PIN_CONF_TypeDef USART_Pin_Config; /* holds pins settings */
+	uint8_t* USART_TxBuffer;
+	uint8_t* USART_RxBuffer;
+	uint32_t TxLength;
+	uint32_t RxLength;
+	uint8_t USART_TxState;
+	uint8_t USART_RxState;
+}USART_Handler_TypedDef;
+
+/* WORD LENGTH */
+#define USART_WORD_LENGTH_8			0
+#define USART_WORD_LENGTH_9			1
+
+/* WAKEUP METHOD */
+#define USART_WAKE_IDLE_LINE		0
+#define USART_WAKE_ADDRESS_MARK		1
+
+/* STOP BIT */
+#define USART_STOP_BIT_1			0
+#define USART_STOP_BIT_0_5			1
+#define USART_STOP_BIT_2			2
+#define USART_STOP_BIT_1_5			3
+
+/* PARITY */
+#define USART_PARITY_DISABLE	DISABLE
+#define USART_PARITY_ENABLE		ENABLE
+
+/* PARITY SELECTION */
+#define USART_PARITY_EVEN			0
+#define USART_PARITY_ODD			1
+
+/* USART MODE */
+#define USART_MODE_TX				0
+#define USART_MODE_RX				1
+#define USART_MODE_TXRX				2	//USART synchronous mode ?
```

Comment:
Syncronous mode is a very specif part, we can use for SPI (but we already have a SPI specific Peripheral), IrDA, LIN, and other like MODBUS DE. All these would need a second driver to implement the protocol.